src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -26,6 +36,81 @@ const ChannelListScreenV2 = () => {
     navigation.navigate('ContactScreen');
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  React.useEffect(() => {
+    const checkNotif = async () => {
+      if (!navigation.isFocused()) return;
+      PushNotification.checkPermissions(
+        (
+          data: PushNotificationPermissions & {
+            authorizationStatus?: number;
+          }
+        ) => {
+          const lastPromptTime = StorageUtils.lastPromptNotification.get();
+          const promptInterval = 48 * 60 * 60 * 1000;
+          const currentTime = new Date().getTime();
+          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+            return; // Don't show the prompt if the interval has not passed
+          }
+
+          const needsPermission =
+            data.alert === false || data.badge === false || data.sound === false;
+
+          const showAlert = (onPressAction: () => void) =>
+            Alert.alert(
+              "Don't Miss New Messages",
+              `Allow notifications to know 
+when friends send you messages.`,
+              [
+                {text: 'Not now', onPress: () => console.log('Cancel Pressed')},
+                {text: 'Allow', onPress: onPressAction, isPreferred: true}
+              ]
+            );
+
+          switch (data.authorizationStatus) {
+            case PERMISSION_STATUS_ACCEPTED:
+              if (needsPermission) {
+                // THIS CASE IT'S TRIGGERED WHEN USER MANUALLY DISABLED THE NOTIFICATION
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              } else {
+                requestPermission();
+              }
+              break;
+            case PERMISSION_STATUS_BLOCKED:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              }
+              break;
+            case PERMISSION_STATUS_PENDING:
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => requestPermission());
+              break;
+            default:
+              break;
+          }
+        }
+      );
+    };
+
+    checkNotif();
+  }, [navigation?.isFocused?.()]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `requestPermission` function and the `useEffect` hook that checks notification permissions are both making asynchronous operations. However, there is no error handling in place for these operations. It's important to handle potential errors to prevent the application from crashing unexpectedly.

For instance, the `messaging().requestPermission()` call could fail due to various reasons such as network issues or Firebase configuration problems. Similarly, the `messaging().getToken()` call could also fail. 

In the `useEffect` hook, the `PushNotification.checkPermissions` function could potentially fail, and the `openSettings().catch(() => console.warn('cannot open settings'))` call might not be sufficient to handle all possible errors.

Here's how you can add error handling:

```diff
  const requestPermission = async () => {
    try {
      const authStatus = await messaging().requestPermission();
      const enabled =
        authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
        authStatus === messaging.AuthorizationStatus.PROVISIONAL;

      if (enabled) {
        const fcmToken = await messaging().getToken();
        const payload = {
          fcm_token: fcmToken
        };
        fcmTokenService(payload);
      }
    } catch (error) {
      console.error("Error requesting permission: ", error);
    }
  };

...

  React.useEffect(() => {
    const checkNotif = async () => {
      try {
        if (!navigation.isFocused()) return;
        PushNotification.checkPermissions(
          (
            data: PushNotificationPermissions & {
              authorizationStatus?: number;
            }
          ) => {
            // ... existing code ...
          }
        );
      } catch (error) {
        console.error("Error checking permissions: ", error);
      }
    };

    checkNotif();
  }, [navigation?.isFocused?.()]);
```

Remember to replace `console.error` with your preferred method of error logging or handling.


<!-- This is an auto-generated comment by OSS CodeRabbit -->